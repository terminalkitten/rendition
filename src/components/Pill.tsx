import * as React from 'react';
import { compose } from 'recompose';
import { PillProps } from 'rendition';
import styled, { withTheme } from 'styled-components';
import asRendition from '../asRendition';
import { getColor, withProps } from '../utils';
import Txt from './Txt';

const BasePill = withProps<PillProps>()(styled(Txt))`
	display: inline-flex;
	align-items: center;
	justify-content: center;
	padding: 2px 8px;
	border-radius: 10px;
	font-family: ${props => props.theme.font};
	font-size: 12px;
	font-weight: 500;
	font-style: normal;
	font-stretch: normal;
	line-height: 14px;
	letter-spacing: 0.5px;
	text-align: left;
	color: #fff;
	background-color: ${props =>
		getColor(props, 'bg', props.shade || 'main') || props.bg || '#7e8085'};
`;

const SmallPill = BasePill.extend`
	padding: 1.4px 5.6px;
	font-size: 8.4px;
	line-height: 11px;
`;

const Pill = ({ small, children, ...props }: PillProps) => {
	const PillComponent = !small ? BasePill : SmallPill;
	return <PillComponent {...props}>{children}</PillComponent>;
};

export default compose(withTheme, asRendition)(Pill) as React.ComponentClass<
	PillProps
>;
