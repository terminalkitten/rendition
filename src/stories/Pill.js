import * as React from 'react'
import { storiesOf } from '@storybook/react'
import withReadme from 'storybook-readme/with-readme'
import styled from 'styled-components'
import { Box, Pill } from '../'
import * as Readme from './README/Pill.md'

const Container = styled.div`
  margin: 30px;
`

storiesOf('Core/Pill', module)
  .addDecorator(withReadme(Readme))
  .addDecorator(story => <Container>{story()}</Container>)
  .add('Standard', () => {
    return (
      <React.Fragment>
        <Box mb={2}>
          <Pill mr={2} primary>
            Primary
          </Pill>
          <Pill mr={2} secondary>
            Secondary
          </Pill>
          <Pill mr={2} tertiary>
            Tertiary
          </Pill>
          <Pill mr={2} quarternary>
            Quarternary
          </Pill>
          <Pill mr={2}>Default</Pill>
        </Box>
        <Box>
          <Pill mr={2} danger>
            Danger
          </Pill>
          <Pill mr={2} warning>
            Warning
          </Pill>
          <Pill mr={2} success>
            Success
          </Pill>
          <Pill mr={2} info>
            Info
          </Pill>
          <Pill bg='purple' color='yellow'>
            Custom
          </Pill>
        </Box>
      </React.Fragment>
    )
  })
  .add('Small', () => {
    return (
      <React.Fragment>
        <Box mb={2}>
          <Pill small mr={2} primary>
            Primary
          </Pill>
          <Pill small mr={2} secondary>
            Secondary
          </Pill>
          <Pill small mr={2} tertiary>
            Tertiary
          </Pill>
          <Pill small mr={2} quarternary>
            Quarternary
          </Pill>
          <Pill small mr={2}>
            Default
          </Pill>
        </Box>
        <Box>
          <Pill small mr={2} danger>
            Danger
          </Pill>
          <Pill small mr={2} warning>
            Warning
          </Pill>
          <Pill small mr={2} success>
            Success
          </Pill>
          <Pill small mr={2} info>
            Info
          </Pill>
          <Pill small bg='purple' color='yellow'>
            Custom
          </Pill>
        </Box>
      </React.Fragment>
    )
  })
