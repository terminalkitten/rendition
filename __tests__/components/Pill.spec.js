/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import Pill from '../../src/components/Pill'

test('Primary Pill renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Pill primary>pill</Pill>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Small Danger Pill renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Pill danger small>pill</Pill>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
