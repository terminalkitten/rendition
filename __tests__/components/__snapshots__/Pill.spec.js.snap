// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Pill renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <div
    className="sc-fjdhpX iBNkjm sc-VigVT bUnCod sc-htoDjs gDCMNh sc-gzVnrw faVbst"
  >
    pill
  </div>
</div>
`;

exports[`Small Danger Pill renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <div
    className="sc-fjdhpX iBNkjm sc-jTzLTM cufzYt sc-htoDjs gDCMNh sc-gzVnrw faVbst"
  >
    pill
  </div>
</div>
`;
